File: 119.pascals-triangle-ii.cpp
/*
 * @lc app=leetcode id=119 lang=cpp
 *
 * [119] Pascal's Triangle II
 */

#include <vector>

using namespace std;

// @lc code=start
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<int> res(rowIndex + 1, 1);
        if (!rowIndex) {
            return res;
        }
        auto row = res;
        for (int i = 0; i < rowIndex; ++i) {
            for (int j = 1; j <= i; ++j) {
                row[j] = res[j - 1] + res[j];
            }
            res.swap(row);
        } 
        return res;
    }
};
// @lc code=end

File: 119.pascals-triangle-ii.py
#
# @lc app=leetcode id=119 lang=python3
#
# [119] Pascal's Triangle II
#

# @lc code=start
from typing import List


class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        if not rowIndex:
            return [1]
        prev = [1]
        for i in range(0, rowIndex):
            row = [1]
            for j in range(i):
                row.append(prev[j] + prev[j + 1])
            row.append(1)
            prev = row
        return prev
# @lc code=end

